OutOfSchool/OutOfSchool.WebApi/Services/BlockedProviderParentService.cs

```diff
@@ -124,30 +129,39 @@ public async Task<Result<BlockedProviderParentDto>> Unblock(BlockedProviderParen
         currentBlock.DateTimeTo = DateTime.Now;
         currentBlock.UserIdUnblock = userId;
 
-        var entity = await blockedProviderParentRepository.UnBlock(mapper.Map<BlockedProviderParent>(currentBlock)).ConfigureAwait(false);
+        var entity = await blockedProviderParentRepository.UnBlock(currentBlock).ConfigureAwait(false);
+        var unblockedParentUserId = Guid.Parse(entity.Parent.UserId);
+        var additionalData = new Dictionary<string, string>()
+            {
+                { ProviderIdKey, entity.ProviderId.ToString() },
+                { ProviderFullTitleKey, entity.Provider.FullTitle },
+                { ProviderShortTitleKey, entity.Provider.ShortTitle },
+            };
+        await notificationService.Create(
+            NotificationType.Parent,
+            NotificationAction.ProviderUnblock,
+            unblockedParentUserId,
+            this,
+            additionalData).ConfigureAwait(false);
 
         return Result<BlockedProviderParentDto>.Success(mapper.Map<BlockedProviderParentDto>(entity));
     }
 
     /// <inheritdoc/>
     public async Task<BlockedProviderParentDto> GetBlock(Guid parentId, Guid providerId)
     {
-        var currentBlock = await blockedProviderParentRepository.GetByFilter(
-            b => b.ParentId == parentId
-                 && b.ProviderId == providerId
-                 && b.DateTimeTo == null).ConfigureAwait(false);
-
-        return mapper.Map<BlockedProviderParentDto>(currentBlock.FirstOrDefault());
+        var currentBlock = await blockedProviderParentRepository
+            .GetBlockedProviderParentEntities(parentId, providerId)
+            .FirstOrDefaultAsync()
+            .ConfigureAwait(false);
+        return mapper.Map<BlockedProviderParentDto>(currentBlock);
     }
 
-    public async Task<bool> IsBlocked(Guid parentId, Guid providerId)
+    public Task<bool> IsBlocked(Guid parentId, Guid providerId)
     {
-        var currentBlock = await blockedProviderParentRepository.GetByFilter(
-            b => b.ParentId == parentId
-                 && b.ProviderId == providerId
-                 && b.DateTimeTo == null).ConfigureAwait(false);
-
-        return currentBlock.Any();
+        return Task.FromResult(blockedProviderParentRepository
```

Comment:
(Concurrency and performance) Please use `Task.FromResult` **only** for synchronously available data (like in `GetNotificationsRecipientIds()` below). Try to find better solution for asynchronous data, usually it is possible. I.e.:
```
    public Task<bool> IsBlocked(Guid parentId, Guid providerId)
        => blockedProviderParentRepository
            .GetBlockedProviderParentEntities(parentId, providerId)
            .AnyAsync();
```

Reasons:
- you don't create another Task object (lower GC/memory pressure), 
- you move decision when to await for a task upper by call-stack (to the client code that knows if some tasks could be parallelized), 
- you don't fake client expectations - that result type `Task` usually means that real (cpu/io-heavy) job could be executed later, but for `Task.FromResult` case we wait the result here and now

Reply:
I've changed it. Thank you for your detailed clarification.